main.go

```diff
@@ -1,93 +1,107 @@
 package main
 
 import (
-    "time"
-    "net/http"
-    "io/ioutil"
-    "sync"
-    "bufio"
-    "fmt"
-    "log"
-    "os"
+	"bufio"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
 )
 
-func seedColumnIds() []string{
-    columnIds := make([]string, 0)
-    f, err := os.Open("columnIds.txt")
+func seedColumnIds() []string {
+	columnIds := make([]string, 0)
+	f, err := os.Open("columnIds.txt")
 
-    if err != nil {
-        log.Fatal(err)
-    }
+	if err != nil {
+		log.Fatal(err)
+	}
 
-    defer f.Close()
+	defer f.Close()
 
-    scanner := bufio.NewScanner(f)
+	scanner := bufio.NewScanner(f)
 
-    for scanner.Scan() {
+	for scanner.Scan() {
 
-        //fmt.Println(scanner.Text())
-        columnIds = append(columnIds, scanner.Text())
-    }
+		//fmt.Println(scanner.Text())
+		columnIds = append(columnIds, scanner.Text())
+	}
 
-    if err := scanner.Err(); err != nil {
-        log.Fatal(err)
-    }
-    return columnIds
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	return columnIds
 }
 
 func main() {
-    var wg sync.WaitGroup
-    work := make(chan string, 10000)
-
-    var columnIdsToDelete []string = seedColumnIds()
-    fmt.Println(len(columnIdsToDelete))
-
-    // write columnIds to the work channel, blocking until a worker goroutine
-    // is able to start work
-    for _, columnId := range columnIdsToDelete {
-        fmt.Println(columnId)
-        work <- columnId
-    }
-
-    // startup pool of 10 go routines and read columnIds from work channel 
-    for {
-        for i := 0; i<=10; i++ {
-            wg.Add(1)
-            go func(w chan string) {
-                defer wg.Done()
-                columnId := <-w
-                url := fmt.Sprintf("https://api.honeycomb.io/1/columns/vector_multitenant_prod_filtered/%s", columnId)
-                sendRequest(url)
-                time.Sleep(1*time.Second)
-                fmt.Println(url)
-            }(work)
-        }
-    }
-
-    wg.Wait()
-
-    fmt.Println("Done")
+	var wg sync.WaitGroup
+	work := make(chan string)
+
+	var columnIdsToDelete []string = seedColumnIds()
+	fmt.Println(len(columnIdsToDelete))
+
+	// write columnIds to the work channel, blocking until a worker goroutine
+	// is able to start work
+	go func(ids []string) {
```

Comment:
Adding the items to the channel has to be in a goroutine, otherwise it doesn't block but just adds all the items to the buffer before any of the receiver goroutines have been created. By putting this in a goroutine, it actually _does_ block and we can get rid of the buffer.